XenServerHealthCheck/ServerListHelper.cs

```diff
@@ -267,5 +267,24 @@ public void UpdateProxy(string proxy)
                 log.Error("Error parsing 'ProxySetting' from XenCenter", e);
             }
         }
+
+        public void UpdateXenCenterMetadata(string message)
+        {
+            log.Info("Receive XenCenter metadata update message");
+
+            try
+            {
+                string[] metadata = message.Split(SEPARATOR);

```

Comment:
Can we prove that none of the metadata can contain this SEPARATOR? If not, we will have to escape it or change it somehow.